### DISCUSSION TITLE
Include the Ability to Add Attributes such as ID & Classes for Template Wrappers in Theme Builder

### DISCUSSION THREAD

**Discussion by Daveden2** [elementor/elementor#24583] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Currently, when we apply a single page/post template to our pages and posts, it replaces the `<main id="content" class="site-main">` with a `<div class="elementor-location-single">` which is an accessibility concern because it breaks the functionality of the Skip Links due to the missing ID. See [W3C Success Criterion](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks) and [WebAIM](https://webaim.org/techniques/skipnav/). A similar issue was raised relating to [full-width page layout](https://github.com/elementor/elementor/issues/19215) and the [Hello Theme skiplink functionality](https://github.com/elementor/hello-theme/issues/217).

It would also be nice to dynamically add CSS classes and data-attributes for custom page styling.

### Describe the solution you'd like.

It would be nice if we could add IDs and Class Names to the Template Wrappers in the same place where we can change the HTML Tags or at the very least, the `<main id="content">` should be added to the single page/post template and the skiplink should function properly.

Currently, probably all blog post templates created using Elementor have issues with the skip links. I only noticed it after reading [this bug report](https://github.com/elementor/elementor/issues/24570) recently.

### Describe alternatives you've considered.

I use the Canvas Page Layout and apply my skip links, header, main and footer templates manually.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by m1ga** [comment]
Maybe you can adjust https://github.com/elementor/elementor/issues/4545#issuecomment-1350386517 to do that. It will add a custom ID field into the settings and it will be rendered on the body tag.

**Reply by Daveden2** [reply]
@m1ga yeah I saw that code, which I think is great but it adds the attributes to the body element.

The skip links specifically needs the ID to be added to the `<main>` element (and the `<footer>` element in some cases)

**Reply by m1ga** [reply]
>  which I think is great but it adds the attributes to the body element.

I know, that's what I use it for and I said you'll need to adjust it :wink:  E.g. `$element->get_name()` to find the container in `add_attributes_to_elements($element)`, then `$element->add_render_attribute` to add the ID. Using that to add classes to containers, should work with IDs too. But not used that, sorry.

But of course: should be in the core!
